app/models/bookmark.rb

```diff
@@ -24,6 +85,10 @@ def self.auto_delete_preferences
     )
   end
 
+  def self.select_type(bookmarks_relation, type)
+    bookmarks_relation.select { |bm| bm.bookmarkable_type == type }
```

Comment:
Since we're dealing with a relation, can we add a filter to the relation instead of doing an `Array#select`?

Reply:
Hmm maybe I need to rename this a bit, it's to filter a relation already fetched from the database, using `where` would send a second query here (I am pretty sure, maybe I am wrong!)

Reply:
Normally when you build a relation, it is not executed until certain methods are called on it. From what I'm seeing, `BookmarkQuery#list_all` builds a relation so at this point it isn't executed yet.

Reply:
Alrighty I'll give it a shot :+1:

Reply:
This does not work, I end up with 3 queries of the same thing in UserBookmarkList#preload_polymorphic_associations

![image](https://user-images.githubusercontent.com/920448/162676391-7e2a5705-1a1e-48ee-9627-a45e3f9331cf.png)

Reply:
(Noting that this is just one copy of the query, there are 3 others listed in MiniProfiler for this route mapping to the 3 lines in user_bookmark_list.rb)